src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -619,5 +619,11 @@ private static bool IsEqualsObject(ISymbol member)
 
         public static INamedTypeSymbol TryConstruct(this INamedTypeSymbol type, ITypeSymbol[] typeArguments)
             => typeArguments.Length > 0 ? type.Construct(typeArguments) : type;
+
+        public static bool IsValidTopLevelStatementsType(this INamedTypeSymbol type)
```

Comment:
It feels like it might involve less work to write a helper which returns the `IMethodSymbol` for the "top-level statements entry point" from the given compilation if one exists, and otherwise returns null.